### PR TITLE
Merge main -> 4.x

### PR DESCRIPTION
**Pull request by mwaggett**

### CODE REVIEW

File: lib/r10k/git/rugged/working_repository.rb [hunk 1]
```diff
@@ -117,11 +117,15 @@ def origin
     with_repo { |repo| repo.config['remote.origin.url'] }
   end
 
-  def dirty?
+  def dirty?(exclude_spec=false)
```
Comment (justinstoller) [inline lib/r10k/git/rugged/working_repository.rb]: I forget, is this default supposed to be `true` in 4.x?

Comment (mwaggett) [reply lib/r10k/git/rugged/working_repository.rb]: ooh hmm, not sure, I'll look

Comment (mwaggett) [reply lib/r10k/git/rugged/working_repository.rb]: looks like it is https://github.com/puppetlabs/r10k/blob/4.x/doc/dynamic-environments/configuration.mkd#exclude_spec 
good catch!

Comment (mwaggett) [reply lib/r10k/git/rugged/working_repository.rb]: updated!

**Review by justinstoller — APPROVED** [review]

**Pull request merged**